### ISSUE TITLE
Chatkit not updating

### ISSUE THREAD

**Issue by Ailanz** [pusher/chatkit-client-react#4]
I have the following code:

 ```
<ChatkitProvider
                            instanceLocator={'v1:us1:.....'}
                            tokenProvider={new TokenProvider({
                                url: chat_auth,
                                queryParams: auth
                            })}
                            userId={'andrew'}>
                            <Row>
                                <Col>
                                    <CandidateList handler={this.setOtherUser}/>
                                </Col>
                                <Col xs={9}>

                                    <ChatBox otherUserId={this.state.otherUserId}/>
                                </Col>
                            </Row>

                        </ChatkitProvider>
```

However when the state.otherUserId is modified, not all of the injected props for chatkit is midified. for example, the this.props.chatkit.messages does not get updated, it is still returning the messages of previous this.state.otherUserId.

please assist, thanks!

**Comment by Ailanz** [comment]
@jonathanlloyd @callum-oakley 

After digging around this is because the chatkit data is injected on componentDidMount so it wont re-initialize the data after prop change.

https://github.com/pusher/chatkit-client-react/blob/master/src/with-chatkit-one-to-one.js#L92

is this expected behavior? If so how do I get around this? I just want to load a oneToOne chat with another userId

**Comment by Ailanz** [comment]
my only work around is adding key to ChatkitProvider to force creating a new one but thats not very performant

**Comment by jonathanlloyd** [comment]
Sorry for the delayed response on this - we have a fix ready which should be released soon.

**Comment by jonathanlloyd** [comment]
@Ailanz this issue should now be resolved in v0.1.4
